frontend/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -376,6 +376,18 @@ export class WorkflowEditorComponent implements OnInit, AfterViewInit, OnDestroy
         .filter(region => region.operators.includes(operator))
         .forEach(region => this.updateRegionElement(region.regionElement, region.operators));
     });
+
+    this.executeWorkflowService
+      .getRegionStateStream()
+      .pipe(untilDestroyed(this))
+      .subscribe(region => {
+        const colorMap: Record<string, string> = {
+          ExecutingDependeePortsPhase: "rgba(244,67,54,0.2)",
```

Comment:
You mentioned "no regions will appear in the red state yet", but that is not true. Some regions will actually go through an `ExecutingDependeePortsPhase`, e.g., a region containing a `Hash Join`'s Probe. You can try it on such a workflow. 

Also the red color might not be appropriate as it usually indicates error.